packages/engine-server/src/engineClient.ts

```diff
@@ -59,7 +62,9 @@ type DendronEngineClientOpts = {
   ws: string;
 };
 
-export class DendronEngineClient implements DEngineClient {
+export class DendronEngineClient implements DEngineClient, EngineEvents {
```

Comment:
instead of having the client implement `EngineEvents`, can we add a method so that it returns a class that implements this interface?

eg. `getEngineEventEmitter(): EngineEvents`

Reply:
Replied on the other comment on `[EngineEvents.spec.ts]`.  Given that, I'm not sure this would help?